pageserver/src/tenant/timeline/eviction_task.rs

```diff
@@ -282,5 +286,57 @@ impl Timeline {
                 );
             }
         }
+
+        // imitate the synthetic size calculation done by the consumption_metrics module.
+        // it runs as a per-tenant task, but the eviction_task.rs is per-timeline.
+        // so, some coordination is required.
+        if let Ok(tenant) = crate::tenant::mgr::get_tenant(self.tenant_id, true).await {
+            // The goal is to make the layer accesses that synthetic size calculation would do
+            // at a period slightly below `p.threshold`.
+            // We want them to happen at least every `p.threshold`, so that eviction won't evict them.
+            // We don't want them to happen more often, because they are somewhat expensive (CPU, IOs).
+            let mut g = tenant.eviction_task_synthetic_size.lock().await;
+            match *g {
+                Some(ts) if ts.elapsed() < p.period => { /* no need to run */ }
```

Comment:
Should this be `p.threshold` ?

Reply:
This was a funny one but fixed already in latest pushes.